File: JavaScript solutions/1108. Defanging an IP Address.js
//approach :1
/**
 * @param {string} address
 *@return {string}
 */
var defangIPaddr = function (address) {
  let res = address.replace(/\./g, "[.]");
  return res;
};

//approach :2
/**
 * @param {string} address
 *  @return {string}
 */
var defangIPaddr = function (address) {
  let ans = address
    .split("")
    .map((char) => {
      if (char === ".") {
        return "[.]";
      }
      return char;
    })
    .join("");
  return ans;
};

File: C++ solutions/1108. Defanging an IP Address.cpp
#include<bits/stdc++.h>
using namespace std;

// solution 
class Solution {
public:
    string defangIPaddr(string address) {
       int len=address.size();
       string ans="";
       for(int i=0;i<len;i++){
           if(address[i]!='.') ans+=address[i]; 
           else ans+="[.]";
       }
       return ans;
    }
};